src/Lifecycle.js

```diff
@@ -289,19 +290,35 @@ export function logout() {
         return;
     }
 
-    return MatrixClientPeg.get().logout().then(onLoggedOut,
-        (err) => {
-            // Just throwing an error here is going to be very unhelpful
-            // if you're trying to log out because your server's down and
-            // you want to log into a different server, so just forget the
-            // access token. It's annoying that this will leave the access
-            // token still valid, but we should fix this by having access
-            // tokens expire (and if you really think you've been compromised,
-            // change your password).
-            console.log("Failed to call logout API: token will not be invalidated");
-            onLoggedOut();
-        }
-    );
+    var QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+    Modal.createDialog(QuestionDialog, {
```

Comment:
I don't think this is an appropriate place to be popping up the modal - this is supposed to be the low-level interface to MatrixClient. For example, this will appear *after* the error dialog telling you you've been logged out when your access-token is invalidated.

Please can this go somewhere else and be called when the "Sign out" button is clicked?